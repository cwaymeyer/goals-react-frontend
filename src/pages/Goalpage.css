body {
  background-color: rgb(244, 248, 252);
}

.Goalpage {
  background-color: rgb(244, 248, 252);
}

.orm-display {
  text-align: center;
}

.orm-display h1 {
  font-size: 3rem;
  font-weight: bold;
  margin: 0;
}
