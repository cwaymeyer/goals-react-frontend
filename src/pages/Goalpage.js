import { useState, useEffect } from "react";
import { useNavigate, useLocation } from "react-router-dom";
import GoalsAPI from "../api/api";
import { Container, Button, Grid } from "@mui/material";
import ProgressContext from "../context/progressContext";
import LineChart from "../LineChart";
import NewProgressForm from "../forms/NewProgressForm";
import LoadingSpinner from "../LoadingSpinner";

function Goalpage({ createProgress }) {
  const navigate = useNavigate();
  const location = useLocation();
  const [infoLoaded, setInfoLoaded] = useState(false);
  const [goal, setGoal] = useState(null);
  const [progressData, setProgressData] = useState([]); // progress data used to populate LineChart
  const [latestProgress, setLatestProgress] = useState(null); // keeps track of the latest progress record, to keep records in order and determine current distance from set goal

  /** Load goal data from API */
  useEffect(function loadGoal() {
    async function getGoal() {
      try {
        const goalData = await GoalsAPI.getGoal(location.state.id);
        setGoal(goalData);
        setLatestProgress({ date: +goalData.start_date, orm: 0 });
      } catch (err) {
        console.error("Error loading goal:", err);
        setGoal(null);
      }
      setInfoLoaded(true);
    }
    setInfoLoaded(false);
    getGoal();
  }, []);

  function deleteGoal(goal_id) {
    GoalsAPI.deleteGoal(goal_id);
    navigate("/home");
  }

  if (!infoLoaded) return <LoadingSpinner />;

  return (
    <ProgressContext.Provider
      value={{
        progressData,
        setProgressData,
        latestProgress,
        setLatestProgress,
      }}
    >
      <Container maxWidth="lg">
        <Button
          variant="contained"
          color="secondary"
          onClick={() => navigate("/home")}
          sx={{ position: "absolute", right: 100, mr: 5 }}
        >
          Back
        </Button>
        <Button
          variant="contained"
          color="warning"
          onClick={() => deleteGoal(location.state.id)}
          sx={{ position: "absolute", right: 0, mr: 5 }}
        >
          Delete
        </Button>
        <h1>{goal.name}</h1>
        <Grid container>
          <Grid item xs={8}>
            <LineChart goalData={goal} />
          </Grid>
          <Grid item xs={4}></Grid>
          <Grid item xs={6}>
            <NewProgressForm
              createProgress={createProgress}
              goal_id={location.state.id}
            />
          </Grid>
          <Grid item xs={3} padding={4}>
            <h3>Progress</h3>
            <progress value="65" max="100" />
          </Grid>
        </Grid>
      </Container>
    </ProgressContext.Provider>
  );
}

export default Goalpage;
