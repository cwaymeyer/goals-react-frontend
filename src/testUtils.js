import UserContext from "./context/userContext";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const progress = [
  {
    id: 10,
    goal_id: 1,
    weight: 50,
    reps: 5,
    orm: "56.3",
    date: "1645477469350",
  },
  {
    id: 11,
    goal_id: 1,
    weight: 50,
    reps: 6,
    orm: "58.1",
    date: "1645477669350",
  },
];

const starting_progress = {
  orm: "56.3",
  date: "1645477269350",
};

const latest_progress = {
  orm: "58.1",
  date: "1645477669350",
};

const goal = {
  id: "1",
  name: "Test Goal",
  username: "demoUser",
  start_weight: null,
  target_weight: 100,
  timeline: 3,
  start_date: "1645477269350",
  end_date: "1653163269350",
  progress: progress,
  starting_progress: starting_progress,
  latest_progress: latest_progress,
};

const demoUser = {
  username: "demoUser",
  password: "password",
  goals: [goal],
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

const renderWithRouter = (ui, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return render(ui, { wrapper: BrowserRouter });
};

export {
  UserProvider,
  renderWithRouter,
  goal,
  starting_progress,
  latest_progress,
};
