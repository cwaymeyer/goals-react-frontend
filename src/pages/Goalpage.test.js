import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Goalpage from "./Goalpage";
import { UserProvider } from "../testUtils";
import { createMemoryHistory } from "history";

it("renders without crashing", function () {
  const history = createMemoryHistory();
  history.push("/1");
  render(
    <UserProvider>
      <MemoryRouter history={history}>
        <Goalpage />
      </MemoryRouter>
    </UserProvider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <UserProvider>
      <MemoryRouter>
        <Goalpage />
      </MemoryRouter>
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
