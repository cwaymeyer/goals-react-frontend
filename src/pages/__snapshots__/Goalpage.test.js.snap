// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoadingSpinner"
  >
    <h3>
      Loading ...
    </h3>
  </div>
</DocumentFragment>
`;
